core/capabilities/ccip/oraclecreator/plugin.go

```diff
@@ -284,33 +290,57 @@ func (i *pluginOracleCreator) createReadersAndWriters(
 
 	contractReaders := make(map[cciptypes.ChainSelector]types.ContractReader)
 	chainWriters := make(map[cciptypes.ChainSelector]types.ChainWriter)
-	for _, chain := range i.chains.Slice() {
-		chainSelector, err1 := i.getChainSelector(chain.ID().Uint64())
+	for relayID, relayer := range i.relayers {
+		chainID, err1 := decimal.NewFromString(relayID.ChainID)
 		if err1 != nil {
-			return nil, nil, err1
+			return nil, nil, fmt.Errorf("error parsing chain ID, expected big int: %s %w", relayID.ChainID, err1)
+		}
+
+		chainSelector, err1 := i.getChainSelector(chainID.BigInt().Uint64())
```

Comment:
Maybe it worths checking if `BigInt().IsUint64()` before casting it, even though if there's an overflow we'll get an error about unknown chain selector.

Reply:
Yeah, we're already boxed in with the chain-selectors API mapping `uint64` and not e.g a string representation of a big integer.